### ISSUE TITLE
Enhancement to add additional custom response headers

### ISSUE THREAD

**Issue by infosecn1nja** [EmpireProject/Empire#1028]
I have an idea why in empire there is no add additional custom response headers, for example:
```
(Empire: listeners/http) > info

    Name: HTTP[S]
Category: client_server

Authors:
  @harmj0y

Description:
  Starts a http[s] listener (PowerShell or Python) that uses a
  GET/POST approach.

HTTP[S] Options:

  Name              Required    Value                            Description
  ----              --------    -------                          -----------
  SlackToken        False                                        Your SlackBot API token to communicate with your Slack instance.
  ProxyCreds        False       default                          Proxy credentials ([domain\]username:password) to use for request (default, none, or other).
  KillDate          False                                        Date for the listener to exit (MM/dd/yyyy).
  Name              True        http1                            Name for the listener.
  Launcher          True        powershell -noP -sta -w 1 -enc   Launcher string.
  DefaultDelay      True        5                                Agent delay/reach back interval (in seconds).
  DefaultLostLimit  True        60                               Number of missed checkins before exiting
  WorkingHours      False                                        Hours for the agent to operate (09:00-17:00).
  SlackChannel      False       #general                         The Slack channel or DM that notifications will be sent to.
  DefaultProfile    True        /include/template/isx.php,/wp06 Default communication profile for the agent.
                                /wp-includes/po.php,/wp08/wp-
                                includes/dtcla.php|Mozilla/5.0
                                (Windows; U; MSIE 7.0; Windows
                                NT 5.2) Java/1.5.0_08|Referer:ht
                                tp://www.google.com|Accept:text/
                                xml,application/xml,application/
                                xhtml+xml,text/html;q=0.9,text/p
                                lain;q=0.8,image/png,*/*;q=0.5
  Headers           True        Server:Apache/2.2.26 (Unix)|X Server header response
                                -Powered-By:PHP/5.3.28|Cache-
                                Control:no-cache|Content-
                                Type:text/html|Keep-
                                Alive:timeout=3, max=100
  Host              True        http://172.16.11.137:80          Hostname/IP for staging.
  CertPath          False                                        Certificate path for https listeners.
  DefaultJitter     True        0.0                              Jitter in agent reachback interval (0.0-1.0).
  Proxy             False       default                          Proxy to use for request (default, none, or other).
  UserAgent         False       default                          User-agent string to use for the staging request (default, none, or other).
  StagingKey        True        0B)eovgNXW#iI;w_RLO?-1^3{Fb[*~GY Staging key for initial agent negotiation.
  BindIP            True        0.0.0.0                          The IP to bind to on the control server.
  Port              True        80                               Port for the listener.
  StagerURI         False                      URI for the stager. Must use /download/. Example: /download/stager.php
```

I hope maybe in the future can be added like that, not just custom server version only.

Thank you.

**Comment by OneLogicalMyth** [comment]
You can do this with profiles. Set the server response header as per the option then use profile to add more headers if needed. 

https://bluescreenofjeff.com/2017-03-01-how-to-make-communication-profiles-for-empire/

Does that help?

**Comment by xorrior** [comment]
Resolved in #1047
